### PR TITLE
perf: branch complete/fail job to separate client

### PR DESCRIPTION
**Pull request by benjie**
In `pg`, getting a client is a little expensive, and we already have a long-running client, so rather than keep fetching new clients to mark jobs completed or failed we can push it to this shared client.

In future, we could batch these commands together over a certain (configurable) number of milliseconds, and then complete/fail all these jobs at once using a more efficient single SQL statement. Maybe.

### CODE REVIEW

**Comment by benjie** [comment]
This is broken-ish (I think something's not being cleaned up fully). This was a huge branch but I've merged parts of it to master in 0d34b93, 1d45c7e, 5d7e210, ed27f21, 823cf6f, ac0b5b5, f255461 and 293705b so we can enjoy the stable parts. This branch needs more work and some significant refactoring.

File: sql/000004.sql [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      from parsed_specs
+      where id = job_id and jobs.locked_by = worker_id
+      returning *
+  ) update graphile_worker.job_queues
+      set locked_by = null, locked_at = null
+      from updated_jobs
+      where job_queues.queue_name = updated_jobs.queue_name;
+
+end;
+$$ language plpgsql volatile strict;
```
[the remark this answers is not in the document]
Comment (benjie) [reply sql/000004.sql]: No reason. SQL might be faster.

File: sql/000004.sql [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+create function graphile_worker.complete_jobs(specs json) returns void as $$
+begin
+  with parsed_specs as (
+    select
+      el->>0 as worker_id,
```
[the remark this answers is not in the document]
Comment (benjie) [reply sql/000004.sql]: It’s used on line 43

[the remark this answers is not in the document]
Comment (gregplaysguitar) [reply sql/000004.sql]: Ah right.. I misread that line.

**Comment by benjie** [comment]
Closing this experiment as it's out of date, and I'm not convinced it actually increased our performance.

**Pull request closed without merging**